node/full_client_test.go

```diff
@@ -951,12 +964,14 @@ func TestMempool2Nodes(t *testing.T) {
 	err = node1.Start()
 	require.NoError(err)
 
-	time.Sleep(1 * time.Second)
-
+	defer func() {
+		assert.NoError(node1.Stop())
+	}()
 	err = node2.Start()
 	require.NoError(err)
-
-	time.Sleep(1 * time.Second)
+	defer func() {
+		assert.NoError(node2.Stop())
+	}()
 
 	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
 	defer cancel()
```

Comment:
Part of this PR should be to do a search for context cancels as well. This will help ensure background threads are getting cleaned up for faster shutdowns.